package_policies/M1.md

```diff
@@ -1,14 +1,7 @@
-**M1.** Each xSDK-compatible package must support either the **community GNU Autoconf or CMake
-options**<sup>3</sup> as defined in the following document:
+**M1.** Each xSDK-compatible package must install through Spack.
 
-[xSDK Community Installation Policies: GNU Autoconf and CMake Options](./community-installation-policies.md).
```

Comment:
Does this mean that we are removing the requirements about CMake and Autoconf options from the mandatory policies?

For example, can people now name their CMake options however they want?

Reply:
To add another related question. Does my package need to include two stages: configuration and build? Adding CMake to PLASMA was beneficial in my experience. I don't plan to remove it of course. But the point is to have a portability of configuration. Not to mention all the configuration flags that xSDK policies specifies. Spack doesn't quite have a similar functionality.

Reply:
@balay @balos1
Should we retain the requirement that if using autoconf and cmake, you must support the options as in the document (we also need to add the document in that case). Also, should we recommend using autoconf or cmake (but still optional of course).

Reply:
and should we recommend a configuration stage where relevant.

Reply:
There has been a lot of discussion before about making the the autoconf/cmake policies less prescriptive, specifically in terms of option names, or even removing them. In my opinion, removing these requirements would be a good thing. I know others have expressed the same sentiment. However, some of the installation policies, like 10., are  partially intended to ensure that the spack build is correct/consistent. So, if we remove the installation policies, we may need to evaluate them first and determine if the intent of any of them needs to be expressed directly in M1 and in terms of spack.

[This issue](https://github.com/xsdk-project/xsdk-issues/issues/79) and  [this related issue](https://github.com/xsdk-project/xsdk-issues/issues/71) hold some of the previous discussion on this subject.

Reply:
There has been a lot of discussion before about making the the autoconf/cmake policies less prescriptive, specifically in terms of option names, or even removing them. In my opinion, removing these requirements would be a good thing. I know others have expressed the same sentiment. However, some of the installation policies, like 10., are  partially intended to ensure that the spack build is correct/consistent. So, if we remove the installation policies, we may need to evaluate them first and determine if the intent of any of them needs to be expressed directly in M1 and in terms of spack.

[This issue](https://github.com/xsdk-project/xsdk-issues/issues/79) and  [this related issue](https://github.com/xsdk-project/xsdk-issues/issues/71) hold some of the previous discussion on this subject.

Reply:
My opinion is that the CMake and Autoconf document is too restrictive, it comes from the perspective of having a unified build system outside of spack, but it is not that relevant any more.

Packages should have a build system that is appropriate for the language (CMake, Autoconf, Python setuptools), and the packages should make an honest attempt to follow the best practices and guidelines of the respective environment.

Here are some example of for my opinion:
* CMake uses BUILD_SHARED_LIBS, it is not acceptable to overwrite the variable with a custom, e.g., MAKE_SHARED
* A pile of Makefiles where the user has to manually edit paths to libraries and compilers is not an acceptable build system
* It is totally OK and acceptable to use <package>_ENABLE_CUDA as opposed to the current requirement of XSDK_ENABLE_CUDA

Reply:
I have added some of the phrasing above. I have not dealt with the installation policies aspect as it sounds like that may require more discussion. I will propose an agenda item for the xSDK meeting on March 19th.